File: chronicleFrontend/src/services/mediaService.ts
import type { MediaAsset } from '../data';
import { apiClient } from './apiClient';

type BackendMediaAsset = {
  id: string;
  name: string;
  type: string;
  size: string;
  date: string;
  image: string;
  usageCount: number;
  altStatus: boolean;
  credit: string;
  license: string;
  category: string;
};

export async function getMediaAssets(): Promise<MediaAsset[]> {
  const items = await apiClient<BackendMediaAsset[]>('/api/media');
  return items.map((item) => ({
    id: item.id,
    name: item.name,
    type: item.type,
    size: item.size,
    date: item.date,
    image: item.image,
    usageCount: item.usageCount,
    altStatus: item.altStatus,
    credit: item.credit,
    license: item.license,
    category: item.category,
  }));
}

File: src/Chronicle.Application/Media/Services/MediaService.cs
using Chronicle.Application.Abstractions.Media;
using Chronicle.Application.Abstractions.Persistence;
using Chronicle.Application.Common.Results;
using Chronicle.Application.Media.Dtos;
using Microsoft.EntityFrameworkCore;

namespace Chronicle.Application.Media.Services;

public sealed class MediaService(IAppDbContext dbContext) : IMediaService
{
    public async Task<Result<IReadOnlyList<MediaAssetResponse>>> GetMediaAssetsAsync(CancellationToken ct)
    {
        var items = await dbContext.MediaAssets
            .AsNoTracking()
            .OrderByDescending(x => x.CreatedAt)
            .Select(x => new MediaAssetResponse
            {
                Id = x.Id,
                Name = x.Name,
                Type = x.Type,
                Size = x.SizeLabel,
                Date = x.CreatedAt.ToString("MMM dd, yyyy"),
                Image = x.ImageUrl,
                UsageCount = x.UsageCount,
                AltStatus = x.HasAltText,
                Credit = x.Credit,
                License = x.License,
                Category = x.Category,
            })
            .ToListAsync(ct);

        return Result<IReadOnlyList<MediaAssetResponse>>.Success(items);
    }
}
